packages/compass-aggregations/src/utils/insights.ts

```diff
@@ -0,0 +1,39 @@
+import type { Signal } from '@mongodb-js/compass-components';
+import type { StoreStage } from '../modules/pipeline-builder/stage-editor';
+
+const ATLAS_LINK = 'https://www.mongodb.com/cloud/atlas/lp/search-1';
+
+const SIGNALS = {
+  'atlas-text-regex-usage-in-stage': {
+    id: 'atlas-text-regex-usage-in-stage',
+    title: 'Inefficient text search operator',
+    description:
+      "In many cases, Atlas Search is MongoDB's most efficient full text search option. Convert your query to $search for a wider range of functionality.",
+    learnMoreLink:
+      'https://www.mongodb.com/docs/atlas/atlas-search/best-practices/',
+    learnMoreLabel: 'Atlas Search best practices',
+  },
+  'non-atlas-text-regex-usage-in-stage': {
+    id: 'non-atlas-text-regex-usage-in-stage',
+    title: 'Inefficient text search operator',
+    description:
+      "In many cases, Atlas Search is MongoDB's most efficient full text search option. Connect with Atlas to explore the power of Atlas Search.",
+    learnMoreLink:
+      'https://www.mongodb.com/docs/atlas/atlas-search/best-practices/',
+    learnMoreLabel: 'Atlas Search best practices',
+  },
+} as const;
+
+export const getInsightForStage = (
+  { stageOperator, value }: StoreStage,
+  isAtlas: boolean
```

Comment:
Seems like everywhere this is used you are doing env === 'atlas', I think it would probably make sense to chance this to passing env here and do the check in one place, what do you think?

Also should we consider ADL as isAtlas here too?

Reply:
yea I think that makes sense. I initially thought it was only one place where I was using this but I was wrong and I missed fixing the other one. Will make this change now.